Automattic/wp-calypso
# Jetpack: Allow connection owner to be changed

Fixes #16021.

**Before**

Site / connection ownership is read-only and non-changeable:

<img width="739" alt="screen shot 2018-06-06 at 16 07 28" src="https://user-images.githubusercontent.com/7767559/41047157-d266157e-69a3-11e8-86de-5335f7436712.png">

**After**
Original connection owner can change the connection owner for sites with more than one connected admin user.

Field for changing the connection owner:
<img width="754" alt="screen shot 2018-06-06 at 16 06 48" src="https://user-images.githubusercontent.com/7767559/41047202-eb3408cc-69a3-11e8-965f-afa5f13156f5.png">

Field for changing the connection owner - selected/focused:
<img width="757" alt="screen shot 2018-06-06 at 16 06 58" src="https://user-images.githubusercontent.com/7767559/41047209-ee8a2bb4-69a3-11e8-9ff2-c89683304410.png">

Confirmation dialog:
<img width="531" alt="screen shot 2018-06-21 at 14 04 16" src="https://user-images.githubusercontent.com/8436925/41715441-2537a4b6-755c-11e8-9e27-cae5d5bd0349.png">

Success notice after confirming the transfer:
<img width="392" alt="screen shot 2018-06-21 at 14 04 31" src="https://user-images.githubusercontent.com/8436925/41715444-255ac55e-755c-11e8-82e1-c0fccf72a9b1.png">

## Todo
- [x] Add tracks call
- ~Make the user identity clearer in the selection dropdown. We currently use wpcom display name, which may be ambiguous in some cases. We might want to switch to wpcom email or user ID. Probably fine to do this in a separate PR.~

Some discussion on display name here: p6TEKc-29d-p2 comment 4917. Note that the display name does already fall back to user ID.


## Testing

* Use a dotorg site with Jetpack 6.2 or newer
* Connect the site to your main dotcom account
* Add another dotorg admin user to the dotorg site
* Connect the new user to a different dotcom account
* Type `localStorage.setItem( 'debug', 'calypso:analytics:tracks' )` in your browser dev console while in Calypso.
* You should now be able to ping ownership back and forth between the connected admin users at http://calypso.localhost:3000/settings/manage-connection/{site}. Note that only the current owner can make the change.
* Verify a `calypso_jetpack_connection_ownership_changed` is being recorded, as indicated in the browser dev console.
